birdhouse/components/stac/config/magpie/stac_hooks.py

```diff
@@ -0,0 +1,144 @@
+
+    # note: matchdict reference of Twitcher owsproxy view is used, just so happens to be same name as Magpie
+    service = get_service_matchdict_checked(request)
+    try:
+        # Getting a new session from the request
+        session = get_session_from_other(request.db)
+        children = ru.get_resource_children(service, db_session=session, limit_depth=5)
+        item_res_create = True
+        # find "stac/stac"
+        for childs in children.values():
+            if childs["node"].resource_name == "stac":
+                # find "stac/stac/collections"
+                for stac_child in childs["children"].values():
+                    if stac_child["node"].resource_name == "collections":
+                        # find the id of the resource matching stac/stac/collections/<collection_id>
+                        for collection_child in stac_child["children"].values():
+                            if collection_child["node"].resource_name == collection_id:
+                                collection_res_id = collection_child["node"].resource_id
+                                # find the id of the resource matching stac/stac/collections/<collection_id>/items
+                                for item_child in collection_child["children"].values():
+                                    if item_child["node"].resource_name == "items":
+                                        item_res_id = item_child["node"].resource_id
+                                        item_res_create = False
+                                        break
+
+        # create resource /stac/stac/collections/<collection_id>/items if does not already exist
+        if item_res_create:
+            item_res = ru.create_resource("items", None, Route.resource_type_name, collection_res_id, db_session=session)
+            item_res_id = item_res.json["resource"]["resource_id"]
+
+        # create resource stac/stac/collection/<collection_id>/items/<item_id>
+        # In try catch since Magpie resource are persistent but stac-db could be wiped
+        try:
+            item_res = ru.create_resource(item_id, display_name, Route.resource_type_name, item_res_id, db_session=session)
+        except Exception as exc:
+            LOGGER.warning("Failed creation of the item %s %s", display_name, str(exc), exc_info=exc)
+        session.commit()   
+   
+    except Exception as exc:
+        LOGGER.error("Unexpected error while creating the item %s %s", display_name, str(exc), exc_info=exc)
+    
+    return response
+
+def extract_display_name(links):
+    # type: (JSON array) -> string
+    """
+    Extract THREDD path from a STAC links
+    """
+    display_name = None
+    for link in links:
+        if link['rel'] == 'source':
+            # Example of title `thredds:birdhouse/CMIP6` -> `birdhouse/CMIP6`
+            display_name = link['title'].split(':')[1]
+            break
+
+    return display_name 
```

Comment:
There should be a check here that raises directly if `display_name` was not resolved, to avoid all following computations that will end up in an error regardless when reaching `ru.create_resource`.